### ISSUE TITLE
Getting stuck in the "trusted phone carrier"

### ISSUE THREAD

**Issue by BoseSj** [twitter/communitynotes#78]
I am getting stuck in the 1/3 step. Can you guys let us know the reason, and what might the step we should be taking to resolve this?

![image](https://user-images.githubusercontent.com/58129377/224344450-49b41cad-9176-481c-83f0-639d2bdcbb7d.png)

**Comment by ScottFerreira** [comment]
I saw this as well. I had a VPN running and when I turned it off for a bit, I was able to get through. Not sure if that will be what's happening for you but figured I would mention just in case.

**Comment by BoseSj** [comment]
> I saw this as well. I had a VPN running and when I turned it off for a bit, I was able to get through. Not sure if that will be what's happening for you but figured I would mention just in case.

I am not using any VPN at the first place.

**Comment by ScottFerreira** [comment]
> > I saw this as well. I had a VPN running and when I turned it off for a bit, I was able to get through. Not sure if that will be what's happening for you but figured I would mention just in case.
> 
> I am not using any VPN at the first place.

Ah okay. Do you mind sharing what phone carrier you use?

**Comment by BoseSj** [comment]
I tried on broadband first, then tried using Jio. Both failed.

**Comment by Jamie0** [comment]
Same trying with Three UK!

**Comment by BoseSj** [comment]
😪they should clarify this issue definitely.

**Comment by kaovilai** [comment]
Had google voice number, switched to T-Mobile number and worked.

**Comment by finaldzn** [comment]
I have a french number from Orange, getting stuck in this step

**Comment by TakkieYah** [comment]
Dutch number vodaphone; also stuck. Anywhere to turn to?

**Comment by wanted0** [comment]
Same issue with Poland and nju mobile (a brand of Orange) carrier. Also, why would VPN matter? I understand the phone carrier is related to your phone number and SMS verification, while your internet connection is a completely unrelated thing. You could be using the same carrier's network, but that's not necessarily the case for most people.

**Comment by januszmk** [comment]
> Same issue with Poland and nju mobile (a brand of Orange) carrier

I believe no network from Poland is accepted. I checked Play, Orange, T-Mobile and no one with number from those carriers can get through. Its weird, because recently they announced that Poland is officially supported

**Comment by wanted0** [comment]
> Its weird, because recently they announced that Poland is officially supported

I can confirm -- tried with T-Mobile, same issue.

We can see first hand the difference between Elonvision and Reality.

I searched through Twitter's source code to find any trace of how the "trusted" phone verification is done, but it doesn't seem to be published.

**Comment by januszmk** [comment]
@wanted0 I was able to sign up to the waitlist after changing to Plus number, so looks like there is 1 working network in Poland

**Comment by avipars** [comment]
If you are using a VOIP number, occasionally this is the cause of the problem

**Comment by januszmk** [comment]
@avipars maybe it is occasionally problem for voip number, but main problem is that normal numbers from countries where community notes launched are not accepted

**Comment by dasayan05** [comment]
Same issue -- not "trusted phone carrier". I am on Lebara in UK.

**Comment by camchis** [comment]
Can't sign up with Three either, one of the biggest carriers in the UK... has anyone successfully signed up with a VOIP/eSim number?

**Comment by SoumyaEXE** [comment]
Jio is supported? @twitterbirdwatch

**Comment by wanted0** [comment]
@jbaxter - my apologies for tagging you directly here, but it seems we're running in circles here. There are many genuine GSM operators, especially across Europe, that are not on the "trusted" list of Twitter. There is no mention anywhere how this trusted list of carriers is built or how to modify it. It blocks the ability of many users to contribute to Community Notes.

Perhaps as an employee you could help address this issue to a proper place internally?

**Comment by voxdevis** [comment]
@twitterbirdwatch which operator is "trusted" in Uzbekistan?

**Comment by muayyat** [comment]
Which operators are trusted in Sweden?

**Comment by kennethsequeira** [comment]
Which are trusted carriers in India?

**Comment by shivam-maurya-git** [comment]
@twitter-service This issue is resolved in India, as community notes services started in India now.

**Comment by wanted0** [comment]
> @twitter-service This issue is resolved in India, as community notes services started in India now.

@shivam-maurya-git this issue is not about Community Notes availability in given countries, but about the secrecy of how Twitter/X selects the carriers as "trusted". There is no public information about it anywhere and many genuine carriers in various countries are banned for no good reason.

Have you checked if all major carriers in India are marked as trusted?

**Comment by shivam-maurya-git** [comment]
@wanted0 Yes, I checked.

**Comment by BoseSj** [comment]
Yeah, this issue seems resolved now. As of 2024, I can sign up using just about any carrier, even my broadband. I suppose as a result of Twitter finally supporting the feature in India.

**Comment by wanted0** [comment]
@BoseSj maybe it's solved in India, but as you can see people were reporting the same issues in other countries. And I can assure you that my mobile phone carrier in Poland is still not recognized as trusted. Please reopen.

**Comment by BoseSj** [comment]
Sure, but as you can see maintainers have not been responding much.

**Comment by wanted0** [comment]
Thank you @BoseSj - apologies for the spam that you'll continue getting, but I don't think there is an option to reassign the issue creator to someone else, and creating a separate issue would be waste IMO.

Let's hope that finally somebody at X like @twitterbirdwatch notices this issue and helps solve it completely.

**Comment by EliusHHimel** [comment]
Hope one-day this issue will be solved for all countries.

**Comment by andybrandt** [comment]
Fairly annoying that living in Poland & not having a foreign SIM I can't get through. Why are all Poland's operators "not trusted"?

**Comment by januszmk** [comment]
@andybrandt plus is trusted

**Comment by andybrandt** [comment]
> @andybrandt plus is trusted

Thanks! Now I have to wonder if it is worth it buying a prepaid Plus card for a couple of months just do get access to community notes posting. :D

BTW: does anyone know what makes an operator "trusted" or "untrusted"?

**Comment by januszmk** [comment]
@andybrandt just remember that you cannot change the number back, because you will loose access to CN. but you can hold the activated sim card without paying for long time

**Comment by wanted0** [comment]
> BTW: does anyone know what makes an operator "trusted" or "untrusted"?

Nobody knows, at least in public. So much for the claimed "open source" approach from Twitter/X. It is also worth noting that other tickets are generally getting attention from X employees or @twitterbirdwatch account, this one, despite having the largest number of comments, is quietly ignored by them.

**Comment by majok-majok** [comment]
@twitterbirdwatch Is any provider "trusted" in Uganda and South Sudan? I have tried pretty much all Ugandan networks in vain

**Comment by kennethsequeira** [comment]
> Which are trusted carriers in India?

Looks like Indian carrier support has been added now.

**Comment by entroxx** [comment]
I am from Romania with Vodafone, my Phone Carrier. Still, I am not able to verify my phone number.
![Uploading IMG_1823.jpeg…]()

**Comment by ThatRandomToast** [comment]
I have this issue with a phone number registered with Mobifone. Is there a list of trusted carriers?
![image](https://github.com/twitter/communitynotes/assets/75108295/b4bf9eb1-6c2d-4e51-ad4a-2b066fbe99b9)

**Comment by gitoffmyln** [comment]
This issue appears stuck.  Maybe I can help.

The repository here is "code that reproduces the note scoring/ranking code that X runs in production" (https://communitynotes.x.com/guide/en/under-the-hood/note-ranking-code)

It is
- a reproduction of the prod note scoring/ranking code
- uses the note data, available from the above url

It is not
- administration of CN contributors, including the sign-up process where "trusted phone carrier" is evaluated
- an implementation of CN -- that would be integrated into X

In other words, this Issue is technically out of scope for the code in this repo.  It is very much in-scope for the goals of CN.  

It might be radioactive for Twitter employees.  Commenting will be difficult because it's whole purpose is to keep bad people/bots out, and that involves proprietary considerations and security.  Closing it would be incorrect, because it's a legitimate problem and they may lose the confidence of the community.  Doing nothing is a valid option, but only for so long.


**What problem is X trying to solve with "trusted phone carriers"?**
They need real human contributors who care.  Not bots.  Not people trying to game the system.  Anyone can write API calls to get tonnes of VoIP numbers, and many VoIP carriers aren't validating identity.  If I provide a VoIP number, there's no assurance.  However, if they rely on the phone carrier to vet people, they have some assurance.

So, "trusted phone carriers" is being used as a proxy for "probably trustworthy human contributors", but it's also eliminating wide classes of humans who value privacy, or use affordable phone plans, or who happen to live in a country with lower trust from an American perspective.  The quality of the scoring/ranking results requires a healthy variety of contributors, and the "trusted phone carrier" requirement is almost certainly eliminating good would-be contributors.

How can X vet accounts to only get trustworthy humans who care, without also eliminating whole groups for the wrong reasons?

**Comment by wanted0** [comment]
@gitoffmyln thanks for your valuable comment. I agree with most of it - problem is where else could we file such an issue? I tried reaching out to CN account and Elon himself on X and they never replied. I don't want to spend money on a premium account, just to possibly, remotely, get better visibility and chance for a reply.

It's also been shown here multiple times that their spam prevention system based on carrier trustworthiness is plainly wrong. In Poland there are 4 actual physical carriers (not MVNOs) and they pretty much equal when it comes to privacy policies, etc. Futhermore, there are strong laws that require customer registration with their ID when obtaining the SIM card (even prepaids). However, the obsolete database or whatever X is using, only allows one carrier out of four -- who's not at all special in reality.

So whomever they trusted to do the research is not actually doing a good job and I think that should be in the best interest of X to fix this issue.

I also realize that employees may be unwilling to post here, but I was counting on their good will to at least escalate this internally and have the problem fixed (even silently, without commenting here), as they've done for India, apparently. This would be enough to just close the issue.

**Comment by gitoffmyln** [comment]
Agreed, this is a good place to file the issue.  (I was trying to get ahead of any argument the issue is completely out of scope, not written well.)

You're right, the issue is about "trusted phone carrier" being incomplete and I'd add the lack of transparency.  (Call it something like "supported phone carriers" if that's easier to publish.)

The documentation makes it clear X wants a wide diversity of perspectives.  My concern is the "trusted phone carrier" approach excludes large groups of potential contributors before they can sign-up:
- privacy-aware individuals (mobile phone number is KYC)
- affordable phone plans (You've indicated this should be fine if the physical carriers are complete for each country.  We don't know without transparency on the list, but probably ok.  eg: Freedom Mobile in Canada is new)
- whole countries (Uzbekistan came up earlier)

The part about privacy-aware individuals is surely out-of-scope, because that's a much broader challenge.  It may have to be a limitation for a while.
